File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        String numberAsString = Integer.toString(numberToCheck);

        if (numberAsString.length() == 1) {
            return true;
        }

        int total = numberAsString.chars()
                .reduce(0, (int a, int b) -> a + pow(b, numberAsString.length()));

        return total == numberToCheck;
    }

    int pow(int a, int b) {
        long result = 1;

        for (int i = 0; i < b; i++) {
            result *= a;
        }

        return (int) result;
    }

}

File: solutions/csharp/armstrong-numbers/1/ArmstrongNumbers.cs
public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        if (number < 1) return true;
        
        var str = number.ToString();

        var numDigits = str.Length;

        var sum = str
            .Select(c => (int)Math.Pow(double.Parse(c.ToString()), numDigits))
            .Sum();
        
        return sum == number;
    }
}